src/util/invariant.cpp

```diff
@@ -0,0 +1,154 @@
+/*******************************************************************\
+
+Module:
+
+Author: Martin Brain, martin.brain@diffblue.com
+
+\*******************************************************************/
+
+
+#include "invariant.h"
+
+#include <stdexcept>
+#include <string>
+#include <sstream>
+
+
+#ifdef CPROVER_INVARIANT_PRINT_STACK_TRACE
+#include <iostream>
+#include <stdlib.h>
+#endif
+
+// Backtraces compiler and C library specific
+#include <assert.h>
+#ifdef __GLIBC__
```

Comment:
This doesn't work for mac and alpine: 
https://travis-ci.org/diffblue/cbmc/jobs/231294968#L140